src/record.ts

```diff
@@ -143,8 +150,10 @@ export class Record implements RecordObject {
    *   const apiResponse = data[1];
    * });
    */
-  delete(callback: ChangeCallback) {
-    this.zone_.deleteRecords(this, callback);
+  delete(): Promise<DeleteRecordResponse>;
+  delete(callback: ChangeCallback): void;
+  delete(callback?: ChangeCallback): void|Promise<DeleteRecordResponse> {
+    this.zone_.deleteRecords(this, callback!);
```

Comment:
It might be good to adopt a convention for type naming. Right now there's things like `ZoneAddRecordsResponse` (e.g. `ClassMethodResponse`) and like `GetRecordsResponse` (e.g. `MethodResponse`) and like `DeleteRecordResponse` (e.g. `MethodClassResponse`). I'd vote for `ClassMethodResponse`.